router/router.go

```diff
@@ -125,13 +125,13 @@ type HandleT struct {
 	sourceIDWorkspaceMap                   map[string]string
 	maxDSQuerySize                         int
 
-	backgroundGroup      *errgroup.Group
-	backgroundCtx        context.Context
-	backgroundCancel     context.CancelFunc
-	backgroundWait       func() error
-	generatorLoopStarted chan struct{}
-	lastQueryRunTime     time.Time
-	timeGained           float64
+	backgroundGroup  *errgroup.Group
+	backgroundCtx    context.Context
+	backgroundCancel context.CancelFunc
+	backgroundWait   func() error
+	startEnded       chan struct{}
```

Comment:
It might be useful to have a test with a succession of starts and stops. I suggest running it with the `-race` flag on as well.

Reply:
indirectly we have been doing that in `TestDynamicCluster`, but we were ignoring the intermittent failures and we were accusing the test as flaky